s/start.go

```diff
@@ -111,8 +179,34 @@ func (spec SupervisorSpec) start(parentCtx context.Context, parentName string) (
 		},
 	}
 
+	onStart := func(err startError) {
+		if err != nil {
+			startCh <- err
+		}
+		close(startCh)
```

Comment:
Sorry if I am getting a little lost, but what happens if more than one child reports a start error? Will the start channel be already closed?

Reply:
No, this is a great question, the error from child creation is handled in the `startChildren` function, and as soon as one fails, it start operation, and reports that error up the chain. We don't start the children in parallel, otherwise we cannot guarantee the start order specified in the settings of the supervisor API (`LeftToRight` vs `RightToLeft`)